plugins/module_utils/storage/dell/libraries/provisioning.py

```diff
@@ -0,0 +1,62 @@
+# Copyright: (c) 2024, Dell Technologies
+
+# Apache License version 2.0 (see MODULE-LICENSE or http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+from ansible_collections.dellemc.powerstore.plugins.module_utils.storage.dell \
+    import utils
+
+LOG = utils.get_logger('provisioning')
+
+
+class Provisioning:
+
+    '''Class with shared provisioing operations'''
+
+    def __init__(self, provisioning, module):
+        """
+        Initialize the provisioing class
+        :param provisioing: The provisioing sdk instance
+        :param module: Ansible module object
+        """
+        self.provisioning = provisioning
+        self.module = module
+
+    def get_nas_server(self, nas_server=None):
+        """Get the details of NAS Server of a given Powerstore storage
+        system"""
+
+        try:
+            msg = 'Getting NAS Server details {0}'.format(nas_server)
+            LOG.info(msg)
+            id_or_name = utils.name_or_id(val=nas_server)
+            LOG.info(id_or_name)
+            if id_or_name == "NAME":
+                nas_details = self.provisioning.get_nas_server_by_name(
+                    nas_server_name=nas_server)
+                if nas_details:  # implement in sdk , workaround
+                    nas_details = nas_details[0]['id']
```

Comment:
Instead of returning the nas_details[0]['id'] better return nas_details[0], so we can re-use this method in other modules.